src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Helpers/Parser/BinderInvocation.cs

```diff
@@ -27,5 +28,72 @@ internal sealed record BinderInvocation
                 Location = invocationSyntax.GetLocation()
             };
         }
+
+        private static bool IsCandidateInvocationExpressionSyntax(SyntaxNode node, out InvocationExpressionSyntax? invocationSyntax)
+        {
+            if (node is InvocationExpressionSyntax
+                {
+                    Expression: MemberAccessExpressionSyntax
+                    {
+                        Name.Identifier.ValueText: string memberName
+                    }
+                } syntax && IsCandidateBindingMethodName(memberName))
+            {
+                invocationSyntax = syntax;
+                return true;
+            }
+
+            invocationSyntax = null;
+            return false;
+
+            static bool IsCandidateBindingMethodName(string name) =>
+                IsCandidateMethodName_ConfigurationBinder(name) ||
+                IsCandidateMethodName_OptionsBuilderConfigurationExtensions(name) ||
+                IsValidMethodName_OptionsConfigurationServiceCollectionExtensions(name);
+        }
+
+        private static bool IsCandidateInvocation(IInvocationOperation operation)
+        {
```

Comment:
Nit: could we add an assert checking that some of the `SyntaxNode` filtering conditions hold?